autoemulate/core/metrics.py

```diff
@@ -236,6 +282,134 @@ def __call__(
+    assumes that the predictive distribution is Gaussian.
+
+    Attributes
+    ----------
+    name: str
+        Display name for the metric.
+    maximize: bool
+        Whether higher values are better. False for MSLL (lower is better).
+    """
+
+    name: str = "msll"
+    maximize: bool = False
+
+    def __call__(
+        self,
+        y_pred: OutputLike,
+        y_true: TensorLike,
+        metric_params: MetricParams | None = None,
+    ) -> TensorLike:
+        """Calculate MSLL metric.
+
+        If no training data is provided in `metric_params.y_train`, the mean log loss
+        is computed without standardization.
+
+        Parameters
+        ----------
+        y_pred: OutputLike
+            Predicted outputs. Must be a distribution.
+        y_true: TensorLike
+            True target values.
+        metric_params: MetricParams
+            Metric parameters including: y_train and reduction.
+
+        Returns
+        -------
+        TensorLike
+            Mean Standardized Log Loss (MSLL) score.
+
+        Raises
+        ------
+        ValueError
+            If y_pred is not a distribution.
+        """
+        if not isinstance(y_pred, DistributionLike):
+            raise ValueError(
+                f"MSLL metric requires probabilistic predictions, got {type(y_pred)}. "
+            )
+
+        if not isinstance(y_true, TensorLike):
+            raise ValueError(f"y_true must be a tensor, got {type(y_true)}")
+
+        if metric_params is None:
+            metric_params = MetricParams()
+
+        # Ensure 2D y_true for consistent handling
+        y_true = y_true.unsqueeze(-1) if y_true.ndim == 1 else y_true
+
+        # Compute mean negative log likelihood (also by output dimension if have
+        # Independent distribution)
+        if isinstance(y_pred, Independent):
+            model_nll_output = -y_pred.base_dist.log_prob(y_true).mean(dim=0)
+            model_nll_total = model_nll_output.mean()
+        else:
+            model_nll_output = None
+            model_nll_total = -y_pred.log_prob(y_true).mean()
+
+        # If no training data, return mean log loss
+        if metric_params.y_train is None:
+            if metric_params.reduction == "mean":
+                return model_nll_total
+            if metric_params.reduction == "none":
+                if model_nll_output is None:
+                    msg = (
+                        "Per-output MLL not available for non-Independent "
+                        "distributions."
+                    )
+                    raise ValueError(msg)
+                return model_nll_output
+            msg = (
+                f"Unknown reduction method: {metric_params.reduction}. "
+                "Expected 'mean' or 'none'."
+            )
+            raise ValueError(msg)
+
+        # Ensure 2D y_train for consistent handling
+        y_train_mean = metric_params.y_train.mean(dim=0, keepdim=True).view(1, -1)
```

Comment:
I think we can remove restriction to 2D here to support output data that is arbitrary shape e.g. `(B, *output_shape)`
```suggestion
        # Keep original shape for y_train_mean to match y_true shape
        y_train_mean = metric_params.y_train.mean(dim=0, keepdim=True)
```